### ISSUE TITLE
[Feature Request] Support ignoring front matter sections in Markdown files

### ISSUE THREAD

**Issue by rvesse** [stkb/Rewrap#277]
When using Markdown to author documentation, some documentation processors e.g. Jekyll, make use of a front matter section e.g.

```
---
layout: default
title: Airline
permalink: index.html
---

# Airline

Airline is an annotation-driven Java library for building Command Line Interfaces (CLIs), it supports simple commands all the way through to complex Git style CLIs with groups and user defined command aliases.

Airline aims to reduce the boiler plate code typically associated with CLIs in Java, many common behaviours can be achieved purely with annotations and zero user code.  Let's take a look at an ultra simple example:
```

This is actual example from one of my open source projects - https://github.com/rvesse/airline/blob/master/docs/index.md

This front matter is basically a section of YAML content that is used to provide variables to the documentation processor when generating the final output so the formatting of this section is significant.

If I use rewrap on this file it will re-wrap this section into a single line e.g.

![Incorrectly wrapped front matter](https://user-images.githubusercontent.com/2104864/123291661-36c1ec80-d50a-11eb-9256-2e9a3fef9fb3.png)

Which will then break the documentation processor if I don't manually correct it.

This makes it annoying to rewrap content that wasn't previously consistently wrapped because you either have to remember to manually correct wrapping on the front matter (which I keep forgetting) or select the content excluding the front matter (which makes doing a quick Ctrl+A, Alt+Q a non-starter)

**Comment by stkb** [comment]
Fixed in v1.16.0

**Comment by rvesse** [comment]
@stkb Awesome, thanks for addressing this 👍
